modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -343,422 +206,15 @@ const DomParser = (settings?: DomParserSettings, schema = Schema()): DomParser =
-          }
-
-          node = rootBlock.lastChild;
-          if (node && node.type === 3) {
-            node.value = node.value.replace(endWhiteSpaceRegExp, '');
-          }
-        }
-      };
-
-      // Check if rootBlock is valid within rootNode for example if P is valid in H1 if H1 is the contentEditabe root
-      if (!schema.isValidChild(rootNode.name, rootBlockName.toLowerCase())) {
-        return;
-      }
-
-      while (node) {
-        const next = node.next;
-
-        if (node.type === 3 || (node.type === 1 && node.name !== 'p' &&
-          !blockElements[node.name] && !node.attr('data-mce-type'))) {
-          if (!rootBlockNode) {
-            // Create a new root block element
-            rootBlockNode = createNode(rootBlockName, 1);
-            rootBlockNode.attr(settings.forced_root_block_attrs);
-            rootNode.insert(rootBlockNode, node);
-            rootBlockNode.append(node);
-          } else {
-            rootBlockNode.append(node);
-          }
-        } else {
-          trim(rootBlockNode);
-          rootBlockNode = null;
-        }
-
-        node = next;
-      }
-
-      trim(rootBlockNode);
-    };
-
-    const createNode = (name: string, type: number): AstNode => {
-      const node = new AstNode(name, type);
-      let list;
-
-      if (name in nodeFilters) {
-        list = matchedNodes[name];
-
-        if (list) {
-          list.push(node);
-        } else {
-          matchedNodes[name] = [ node ];
-        }
-      }
-
-      return node;
-    };
-
-    const removeWhitespaceBefore = (node: AstNode): void => {
-      const blockElements = schema.getBlockElements();
-
-      for (let textNode = node.prev; textNode && textNode.type === 3;) {
-        const textVal = textNode.value.replace(endWhiteSpaceRegExp, '');
-
-        // Found a text node with non whitespace then trim that and break
-        if (textVal.length > 0) {
-          textNode.value = textVal;
-          return;
-        }
-
-        const textNodeNext = textNode.next;
-
-        // Fix for bug #7543 where bogus nodes would produce empty
-        // text nodes and these would be removed if a nested list was before it
-        if (textNodeNext) {
-          if (textNodeNext.type === 3 && textNodeNext.value.length) {
-            textNode = textNode.prev;
-            continue;
-          }
-
-          if (!blockElements[textNodeNext.name] && textNodeNext.name !== 'script' && textNodeNext.name !== 'style') {
-            textNode = textNode.prev;
-            continue;
-          }
-        }
-
-        const sibling = textNode.prev;
-        textNode.remove();
-        textNode = sibling;
-      }
-    };
-
-    const cloneAndExcludeBlocks = (input: SchemaMap) => {
-      const output: SchemaMap = {};
-
-      for (const name in input) {
-        if (name !== 'li' && name !== 'p') {
-          output[name] = input[name];
-        }
-      }
-
-      return output;
-    };
-
-    const parser = SaxParser({
```

Comment:
How come `SaxParser` wasn't removed as part of this PR?

Reply:
I've got a task inside this epic to delete the SaxParser.

Reply:
Why not just add the TODO to SaxParser as I really want a TODO in there at the very least?